src/plugins/intel_cpu/src/nodes/executors/dnnl/dnnl_fullyconnected_primitive.cpp

```diff
@@ -0,0 +1,358 @@
+        dnnlpoc.appendDecompressionZeroPoints(key.attrs.decompressionSubtractPtr, !key.attrs.weightsNonTransposed);
+
+    return dnnlpoc.compose();
+}
+
+static primitive_desc createPrimitiveDesc(const dnnl::memory::desc& inputDesc,
+                                          const dnnl::memory::desc& weightDesc,
+                                          const dnnl::memory::desc& biasDesc,
+                                          const dnnl::memory::desc& outputDesc,
+                                          const dnnl::primitive_attr& attr,
+                                          const dnnl::engine& engine,
+                                          const std::vector<impl_desc_type>& implPriorities,
+                                          const bool useSparseWeights,
+                                          const bool useWeightsDecompression);
+
+static VectorDims makeDummyInputDims(const Shape& inShape, const Shape& wShape) {
+    const auto& weightDims = wShape.getStaticDims();
+
+    auto inMinDims = inShape.getMinDims();
+    auto inMaxDims = inShape.getMaxDims();
+    inMinDims.back() = weightDims.back();
+    inMaxDims.back() = weightDims.back();
+
+    return MemoryDescUtils::makeDummyShape(Shape(inMinDims, inMaxDims)).getStaticDims();
+}
+
+static VectorDims makeDummyOutputDims(const VectorDims& inShape, const VectorDims& wShape, const size_t out_rank) {
+    size_t activationRank = inShape.size();
+    size_t channelRank = wShape.size() - 1;
+    // activation   weight    output_shape
+    // NCHW         CoCHW     NCo
+    // TNC          CoC       TNCo
+    // NC           CoC       NCo
+    VectorDims outputShape(out_rank, 1);
+    // set Co
+    outputShape.back() = wShape[0];
+    // set batch dims
+    size_t batchRank = activationRank - channelRank;
+    size_t startIdx = out_rank - batchRank - 1;
+    for (size_t i = 0; i < batchRank; i++) {
+        outputShape[i + startIdx] = inShape[i];
+    }
+
+    return outputShape;
+}
+
+DnnlShapeAgnosticDataPtr DnnlFCPrimitive::createShapeAgnosticData(const FCConfig& key,
+                                                                  const MemoryArgs& memory,
+                                                                  const ExecutorContext::CPtr context,
+                                                                  const bool cacheWeights) {
+    DEBUG_LOG("Creating shape agnostic data");
+    auto srcDesc = key.descs.src[0];
+    const auto& weiDesc = key.descs.src[1];
+    auto dstDesc = key.descs.dst[0];
+
+    if (srcDesc->getShape().isDynamic()) {
+        const auto& inShape = srcDesc->getShape();
+        const auto& wShape = weiDesc->getShape();
+        const auto& inDymmyDims = makeDummyInputDims(inShape, wShape);
+        srcDesc = srcDesc->cloneWithNewDims(inDymmyDims);
+        const auto& outDymmyDims =
+            makeDummyOutputDims(inDymmyDims, wShape.getStaticDims(), dstDesc->getShape().getRank());
```

Comment:
From my understanding even shapeInference instance is not an ultimate solution for this problem.
As you know, there can be additional context, for example the post ops.
Also shape inference does not really help with creation of inputDummyShapes, which still needs to be constructed manually.
If we want to generalize the dummy shape creation, I would say we need a function, like
createDummyConfig(executor::Config<FCAttrs>), which creates dummy Config from partially defined Config.
And yes, the shapeInference logic would be a part of this function, but I am not sure if it is worth passing the shapeInference instance inside the executor.
What do you think?

Reply:
Need to check the current state of the ngraph shape inference implementations. If they may be called just passing all the necessary arguments, may be the dnnl implementations will be able to use them directly. On the other hand, the shape inference object might be a part of the executor context, but only for specific executors, though it would break the general concept of the uniform executor context.

Reply:
Discussed. The decision is to handle the issues with dummy shapes case by case.